### PR TITLE
UNet: Allow to freely define the number of channels per depth in subclasses

### PR DESCRIPTION
**Pull request by mys007**
This PR proposes to replace the hard-coded definition of the number of channels `2**depth` by function `_get_num_channels()` which can be overridden in subclasses, e.g. as `return list_of_widths[depth - 1]`. In addition, `pre_conv_op_regularizer_factory` and `post_conv_op_regularizer_factory` are removed, as they seem unused.

### CODE REVIEW

File: inferno/extensions/models/unet.py [hunk 1]
```diff
@@ -119,14 +119,18 @@ def __init__(self, in_channels, dim, out_channels=None, depth=3,
         assert len(self.n_channels_per_output) == self._store_conv_down.count(True) + \
             self._store_conv_up.count(True) + int(self._store_conv_bottom)
 
+    def _get_num_channels(self, depth):
```
Comment (DerThorsten) [inline inferno/extensions/models/unet.py]: we could even go  for `get_num_channels` without a underscore

**Comment by mys007** [comment]
Cool, what kind of tests do you suggest?

**Comment by DerThorsten** [comment]
I think we should have a  custom unet in the tests which overwrites `_get_num_channels(self, depth)`
with some corner cases, maybe a constant number of channels / only a single of channel, number of channels which decrease when we go down in the unet etc,
Using the side-output one could even check if the layers indeed return the right number of channels for a certain depth.

**Comment by DerThorsten** [comment]
btw thanks for the PR, this was indeed some missing functionality

**Comment by DerThorsten** [comment]
@mys007 If you think that these additional tests are a bit overkill, we could also merge now and extend the tests later

**Comment by constantinpape** [comment]
LGTM from my side.

File: inferno/extensions/models/unet.py (hunk 1 above)
Comment (DerThorsten) [reply inferno/extensions/models/unet.py]: @constantinpape  what do you think here?

**Comment by mys007** [comment]
Thanks a lot for the review. I will add the tests on Monday!

**Comment by mys007** [comment]
Tests have been added.

**Comment by DerThorsten** [comment]
once traivs is done i'll merge it,
thanks for the tests and the PR
